File: codes/csharp/utils/ListNode.cs
// File: ListNode.cs
// Created Time: 2022-12-16
// Author: mingXta (1195669834@qq.com)

namespace hello_algo.utils;

/* Definition for a singly-linked list node */
public class ListNode {
    public int val;
    public ListNode? next;

    public ListNode(int x) {
        val = x;
    }

    /* Generate a linked list with an array */
    public static ListNode? ArrToLinkedList(int[] arr) {
        ListNode dum = new ListNode(0);
        ListNode head = dum;
        foreach (int val in arr) {
            head.next = new ListNode(val);
            head = head.next;
        }
        return dum.next;
    }

    /* Get a list node with specific value from a linked list */
    public static ListNode? GetListNode(ListNode? head, int val) {
        while (head != null && head.val != val) {
            head = head.next;
        }
        return head;
    }

    public override string? ToString() {
        List<string> list = new();
        var head = this;
        while (head != null) {
            list.Add(head.val.ToString());
            head = head.next;
        }
        return string.Join("->", list);
    }
}

File: codes/dart/utils/list_node.dart
/**
 * File: list_node.dart
 * Created Time: 2023-01-23
 * Author: Jefferson (JeffersonHuang77@gmail.com)
 */

/**
 * Definition for a singly-linked list node
 */
class ListNode {
  int val;
  ListNode? next;

  ListNode(this.val, [this.next]);

  @override
  String toString() {
    return 'ListNode{val: $val, next: $next}';
  }
}

/* Generate a linked list with a vector */
ListNode? listToLinkedList(List<int> list) {
  ListNode dum = ListNode(0);
  ListNode? head = dum;
  for (int val in list) {
    head?.next = ListNode(val);
    head = head?.next;
  }
  return dum.next;
}
